packages/jest-worker/src/workers/ChildProcessWorker.ts

```diff
@@ -44,19 +47,30 @@ import {
 export default class ChildProcessWorker implements WorkerInterface {
   private _child!: ChildProcess;
   private _options: WorkerOptions;
-  private _onProcessEnd!: OnEnd;
-  private _fakeStream: PassThrough | null;
+
   private _request: ChildMessage | null;
   private _retries!: number;
-  private _stderr: ReturnType<typeof mergeStream> | null;
+  private _onProcessEnd!: OnEnd;
+
+  private _fakeStream: PassThrough | null;
   private _stdout: ReturnType<typeof mergeStream> | null;
+  private _stderr: ReturnType<typeof mergeStream> | null;
+
+  private _exitPromise: Promise<void>;
+  private _resolveExitPromise!: () => void;
```

Comment:
why the TS override `!` here (and in a few places around it)?

Reply:
TS doesn't know that the function argument to `new Promise` is invoked immediately, so it doesn't know by itself that this is guaranteed to be initialized. [Relevant TS issue](https://github.com/Microsoft/TypeScript/issues/11498)